src/contexts/AppProvider.jsx

```diff
@@ -6,7 +6,7 @@ const AppProvider = ({ children }) => {
     const [users, setUsers] = useState([]);
     const [currentUser, setCurrentUser] = useState(() => {
       const savedUser = sessionStorage.getItem('currentUser');
-      return savedUser ? JSON.parse(savedUser) : {};
+      return savedUser === "undefined" ? {} : JSON.parse(savedUser);
```

Comment:
`sessionStorage` only contains `string` values, so we have to explicity check for`"undefined"` string here. 

this is a common pitfall when working with `sessionStorage`. alternatively to eliminate confusion, you could do

```
// App.jsx L157
sessionStorage.setItem('currentUser', JSON.stringify(currentUser) ?? "unavailable");
```

and the checking for this line would become

```
return savedUser === "unavailable" ? {} : JSON.parse(savedUser);
```

which would prevent the `undefined` vs `"undefined"` confusion